### PR TITLE
Add services specs: GitHub Consumers

### PR DESCRIPTION
**Pull request by ImMPrada**
```
Github::ApiClient
  #fetch_profile_data
    when the profile exists on github
      returns the response code
      returns the response body
    when the profile doesn't exist on github
      returns the error code
      returns the a nil body
    when github api is shutdown
      returns the error code
      returns the a nil body
  #fetch_profile_repos
    when the profile exists on github
      returns the response code
      returns the response body
    when the profile doesn't exist on github
      returns the error code
      returns the a nil body
    when github api is shutdown
      returns the error code
      returns the a nil body
  #fetch_repo_languages
    when the profile, and a repo exists on github
      returns the response code
      returns the response body
    when the profile doesn't exist on github
      returns the error code
      returns the a nil body
    when github api is shutdown
      returns the error code
      returns the a nil body
```

```
Github::ProfileConsumer
  when the profile exists on github, ApiClient code: 200
    returns right body class
    returns right body content
  when ApiClient code: 4XX
    returns right body class
    returns right body content
```

```
Github::ReposConsumer
  when the profile exists on github, ApiClient code: 200
    returns right body class
    returns right body content
  when ApiClient code: 4XX
    returns right body class
    returns right body content
```

```
Github::RepoLanguagesConsumer
  when ApiClient code: 200
    returns right body class
    returns right body content
  when ApiClient code: 4XX
    returns right body class
    returns right body content
```

### CODE REVIEW

File: spec/services/github/profile_consumer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        Github::ApiClient,
+        fetch_profile_data: {},
+        code: 403,
+        body: nil
+      )
+      allow(Github::ApiClient).to receive(:new).and_return(profile_response)
+    end
+
+    it 'returns right body class' do
+      expect(profile_consumer.call.class).to eq(NilClass)
```
Comment (mapra99) [inline spec/services/github/profile_consumer_spec.rb:77]: you can DRY this `profile_consumer.call` line moving it to a variable helper using `let`
```ruby
let(:result) { profile_consumer.call }
```
```suggestion
      expect(result.class).to eq(NilClass)
```

File: spec/services/github/repos_consumer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        fetch_profile_repos: {},
+        code: 403,
+        body: nil
+      )
+      allow(Github::ApiClient).to receive(:new).and_return(profile_response)
+    end
+
+    it 'returns right body class' do
+      expect(repos_consumer.call.class).to eq(NilClass)
+    end
```
Comment (mapra99) [inline spec/services/github/repos_consumer_spec.rb:62]: Maybe this is better
```suggestion
    it 'returns nil' do
      expect(repos_consumer.call).to eq(nil)
    end
```

**Review by mapra99 — APPROVED** [review]
🚀

File: spec/services/github/profile_consumer_spec.rb (hunk 1 above)
Comment (ImMPrada) [reply spec/services/github/profile_consumer_spec.rb:77]: I like it

File: spec/services/github/repos_consumer_spec.rb (hunk 2 above)
Comment (ImMPrada) [reply spec/services/github/repos_consumer_spec.rb:62]: I like it

`expect(result).to be_nil`

**Pull request merged**